scalyr_agent/builtin_monitors/docker_monitor.py

```diff
@@ -596,14 +692,23 @@ def __get_last_request_for_log( self, path ):
 
         return scalyr_util.seconds_since_epoch( result )
 
-    def __create_log_config( self, parser, path, attributes, parse_as_json=False ):
+    def __create_log_config( self, parser, path, attributes, base_config={}, parse_as_json=False ):
         """Convenience function to create a log_config dict from the parameters"""
 
-        return { 'parser': parser,
-                 'path': path,
-                 'parse_lines_as_json' : parse_as_json,
-                 'attributes': attributes
-               }
+        result = base_config.copy()
+
+        if 'parser' not in result and 'parser' not in attributes:
+            result['parser'] = parser
+
+        result['path'] = path
+        result['parse_lines_as_json'] = parse_as_json
+
+        if 'attributes' in result:
+            result['attributes'].update( attributes )
+        else:
+            result['attributes'] = JsonObject( attributes )
```

Comment:
Why does this have to be a `JsonObject`?  Can you add a comment.

Reply:
Done.